### PR TITLE
bgp - Add support for 'replaced' and 'overridden' states

### PR DESCRIPTION
**Pull request by ArunSaravananBalachandran**
##### SUMMARY

- Add support for replaced, overridden states in `bgp` module.
- Add regression and unit test cases to cover replaced and overridden states.

##### GitHub Issues
List the GitHub issues impacted by this PR. If no Github issues are affected, please indicate this with "N/A".

| GitHub Issue # |
| -------------- |
| N/A |

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

- sonic_bgp

##### Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.
- [x] I have provided a summary for this PR in valid "fragment" file format in the "changelogs/fragments" directory of this repository branch. Reference : [Ansible Change Log Document](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

##### How Has This Been Tested?

- Regression test HTML report:
[bgp_regression_report-2023-05-12-13-52-27.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/11462994/bgp_regression_report-2023-05-12-13-52-27.pdf)

- Regression test detailed log:
[bgp_regression_terminal_logs_12_05_2023.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/11462996/bgp_regression_terminal_logs_12_05_2023.txt)

### CODE REVIEW

File: plugins/module_utils/network/sonic/config/bgp/bgp.py [hunk 1]
```diff
@@ -128,19 +134,85 @@ def set_state(self, want, have):
...
+            requests = del_requests
+
+        add_commands = get_diff(want, have, TEST_KEYS)
+        if add_commands:
+            for command in add_commands:
+                as_val = command['bgp_as']
+                vrf_name = command['vrf_name']
+
+                # max_med -> on_startup options are modified or deleted at once.
+                # Diff might the reflect the correct commands if only one of
```
Comment (kerry-meyer) [inline plugins/module_utils/network/sonic/config/bgp/bgp.py:168]:
```suggestion
                # Diff might not reflect the correct commands if only one of
```

File: plugins/module_utils/network/sonic/config/bgp/bgp.py [hunk 2]
```diff
@@ -128,19 +134,85 @@ def set_state(self, want, have):
...
+            requests = del_requests
+
+        add_commands = get_diff(want, have, TEST_KEYS)
+        if add_commands:
+            for command in add_commands:
+                as_val = command['bgp_as']
+                vrf_name = command['vrf_name']
+
+                # max_med -> on_startup options are modified or deleted at once.
+                # Diff will not the reflect the correct commands if only one of
```
Comment (kerry-meyer) [inline plugins/module_utils/network/sonic/config/bgp/bgp.py:202]:
```suggestion
                # Diff will not reflect the correct commands if only one of
```

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
The code changes and test results look fine. There are just a couple of minor typos I noticed in comments that should be fixed.

I will approve the PR as soon as those are fixed.

File: plugins/module_utils/network/sonic/config/bgp/bgp.py (hunk 1 above)
Comment (ArunSaravananBalachandran) [reply plugins/module_utils/network/sonic/config/bgp/bgp.py]: Done

File: plugins/module_utils/network/sonic/config/bgp/bgp.py (hunk 2 above)
Comment (ArunSaravananBalachandran) [reply plugins/module_utils/network/sonic/config/bgp/bgp.py]: Done

**Review by kerry-meyer — APPROVED** [review]
Thanks for doing the comment updates.

Approved.

**Comment by ArunSaravananBalachandran** [comment]
Have updated the 'replaced' and 'overridden' implementation to handle the 'rt_delay' option added to bgp module by PR #244

**Review by kerry-meyer — APPROVED** [review]
Thanks for doing the last-minute addition for route map process delay.

All changes and test results look good.

Approved.

**Pull request merged**